src/sentry/integrations/discord/client.py

```diff
@@ -77,9 +115,17 @@ def send_message(
         """
         Send a message to the specified channel.
         """
-        self.post(
-            MESSAGE_URL.format(channel_id=channel_id),
-            data=message.build(notification_uuid=notification_uuid),
-            timeout=5,
-            headers=self.prepare_auth_header(),
-        )
+        if self.is_bot_blocked_by_overrides(channel_id=channel_id):
```

Comment:
if NOT

Reply:
are we going to check this every single time? will this become very expensive?

Reply:
compare to this approach, should we try to post the message and check if the channel is blocked when we get 403?